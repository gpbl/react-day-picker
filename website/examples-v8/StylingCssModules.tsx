import defaultStyles from "react-day-picker-v8/dist/style.module.css";

import { DayPicker } from "./react-day-picker-v8";

export function StylingCssModules() {
  return <DayPicker classNames={{ ...defaultStyles }} />;
}
