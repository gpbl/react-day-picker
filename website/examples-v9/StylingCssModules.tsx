import { DayPicker } from "react-day-picker";
import defaultStyles from "react-day-picker/style.module.css";

export function StylingCssModules() {
  return <DayPicker classNames={{ ...defaultStyles }} />;
}
