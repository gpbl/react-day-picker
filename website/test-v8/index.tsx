import { render, screen } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import MockDate from "mockdate";

export { act, screen } from "@testing-library/react";

export function app() {
  return screen.getByRole("application");
}

export function grid(name?: string) {
  return screen.getByRole("grid", { name });
}

export function columnheader(name?: string) {
  return screen.getByRole("columnheader", { name });
}

export function rowheader(name?: string) {
  return screen.getByRole("rowheader", { name });
}

export function gridcell(date: Date) {
  return screen.getByRole("gridcell", { name: date.getDate().toString() });
}

export function monthDropdown() {
  return screen.getByRole("combobox", { name: "Month:" });
}

export function yearDropdown() {
  return screen.getByRole("combobox", { name: "Year:" });
}

export function renderApp(children: React.ReactNode) {
  return render(<div role="application">{children}</div>);
}

export function prevButton() {
  return screen.getByRole("button", { name: "Go to previous month" });
}

export function nextButton() {
  return screen.getByRole("button", { name: "Go to next month" });
}

export function focusedElement() {
  if (!document.activeElement) {
    throw new Error("Could not find any active element");
  }
  return document.activeElement;
}

export function mockDate(date: Date) {
  beforeAll(() => MockDate.set(date));
  afterAll(() => MockDate.reset());
}

export function getDayButton(day: Date, index = 0) {
  return screen.getAllByRole("gridcell", {
    name: day.getDate().toString()
  })[index];
}

export function getMonthCaption(displayIndex = 0) {
  return screen.getAllByRole("presentation")[displayIndex];
}

export const user = userEvent.setup();
