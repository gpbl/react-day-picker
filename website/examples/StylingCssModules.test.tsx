import { app, mockDate, renderApp } from "../test-v8";

import { StylingCssModules } from "./StylingCssModules";

const today = new Date(2021, 10, 25);
mockDate(today);

beforeEach(() => {
  renderApp(<StylingCssModules />);
});

test("should match the snapshot", () => {
  expect(app()).toMatchSnapshot();
});
