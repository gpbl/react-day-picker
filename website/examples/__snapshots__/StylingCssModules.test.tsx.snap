// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  role="application"
>
  <div
    class="rdp "
  >
    <div
      class="rdp-months"
    >
      <div
        class="rdp-month rdp-caption_start rdp-caption_end"
      >
        <div
          class="rdp-caption"
        >
          <div
            aria-live="polite"
            class="rdp-caption_label"
            id="react-day-picker-1"
            role="presentation"
          >
            November 2021
          </div>
          <div
            class="rdp-nav"
          >
            <button
              aria-label="Go to previous month"
              class="rdp-button_reset rdp-button rdp-nav_button rdp-nav_button_previous"
              name="previous-month"
              type="button"
            >
              <svg
                class="rdp-nav_icon"
                height="16px"
                viewBox="0 0 120 120"
                width="16px"
              >
                <path
                  d="M69.490332,3.34314575 C72.6145263,0.218951416 77.6798462,0.218951416 80.8040405,3.34314575 C83.8617626,6.40086786 83.9268205,11.3179931 80.9992143,14.4548388 L80.8040405,14.6568542 L35.461,60 L80.8040405,105.343146 C83.8617626,108.400868 83.9268205,113.317993 80.9992143,116.454839 L80.8040405,116.656854 C77.7463184,119.714576 72.8291931,119.779634 69.6923475,116.852028 L69.490332,116.656854 L18.490332,65.6568542 C15.4326099,62.5991321 15.367552,57.6820069 18.2951583,54.5451612 L18.490332,54.3431458 L69.490332,3.34314575 Z"
                  fill="currentColor"
                  fill-rule="nonzero"
                />
              </svg>
            </button>
            <button
              aria-label="Go to next month"
              class="rdp-button_reset rdp-button rdp-nav_button rdp-nav_button_next"
              name="next-month"
              type="button"
            >
              <svg
                class="rdp-nav_icon"
                height="16px"
                viewBox="0 0 120 120"
                width="16px"
              >
                <path
                  d="M49.8040405,3.34314575 C46.6798462,0.218951416 41.6145263,0.218951416 38.490332,3.34314575 C35.4326099,6.40086786 35.367552,11.3179931 38.2951583,14.4548388 L38.490332,14.6568542 L83.8333725,60 L38.490332,105.343146 C35.4326099,108.400868 35.367552,113.317993 38.2951583,116.454839 L38.490332,116.656854 C41.5480541,119.714576 46.4651794,119.779634 49.602025,116.852028 L49.8040405,116.656854 L100.804041,65.6568542 C103.861763,62.5991321 103.926821,57.6820069 100.999214,54.5451612 L100.804041,54.3431458 L49.8040405,3.34314575 Z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
        <table
          aria-labelledby="react-day-picker-1"
          class="rdp-table"
          role="grid"
        >
          <thead
            class="rdp-head"
          >
            <tr
              class="rdp-head_row"
            >
              <th
                aria-label="Sunday"
                class="rdp-head_cell"
                scope="col"
              >
                Su
              </th>
              <th
                aria-label="Monday"
                class="rdp-head_cell"
                scope="col"
              >
                Mo
              </th>
              <th
                aria-label="Tuesday"
                class="rdp-head_cell"
                scope="col"
              >
                Tu
              </th>
              <th
                aria-label="Wednesday"
                class="rdp-head_cell"
                scope="col"
              >
                We
              </th>
              <th
                aria-label="Thursday"
                class="rdp-head_cell"
                scope="col"
              >
                Th
              </th>
              <th
                aria-label="Friday"
                class="rdp-head_cell"
                scope="col"
              >
                Fr
              </th>
              <th
                aria-label="Saturday"
                class="rdp-head_cell"
                scope="col"
              >
                Sa
              </th>
            </tr>
          </thead>
          <tbody
            class="rdp-tbody"
          >
            <tr
              class="rdp-row"
            >
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  role="gridcell"
                />
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  1
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  2
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  3
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  4
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  5
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  6
                </div>
              </td>
            </tr>
            <tr
              class="rdp-row"
            >
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  7
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  8
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  9
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  10
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  11
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  12
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  13
                </div>
              </td>
            </tr>
            <tr
              class="rdp-row"
            >
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  14
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  15
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  16
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  17
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  18
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  19
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  20
                </div>
              </td>
            </tr>
            <tr
              class="rdp-row"
            >
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  21
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  22
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  23
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  24
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day rdp-day_today"
                  role="gridcell"
                >
                  25
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  26
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  27
                </div>
              </td>
            </tr>
            <tr
              class="rdp-row"
            >
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  28
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  29
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  class="rdp-day"
                  role="gridcell"
                >
                  30
                </div>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  role="gridcell"
                />
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  role="gridcell"
                />
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  role="gridcell"
                />
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  role="gridcell"
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
